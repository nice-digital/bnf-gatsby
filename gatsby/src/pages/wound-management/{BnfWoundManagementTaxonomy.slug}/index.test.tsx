import { render, screen, within, waitFor } from "@testing-library/react";
import React from "react";

import WoundManagementTaxonomyPage, {
	type WoundManagementTaxonomyPageProps,
} from "./";

const props: WoundManagementTaxonomyPageProps = {
	data: {
		allBnfWoundManagementTaxonomy: {
			taxonomies: [
				{
					title: "Basic wound contact dressings",
					slug: "basic-wound-contact-dressings",
				},
				{
					title: "Advanced wound dressings",
					slug: "advanced-wound-dressings",
				},
			],
		},
		bnfWoundManagementTaxonomy: {
			title: "Advanced wound dressings",
			slug: "advanced-wound-dressings",
			text: "<p> Advanced wound dressings can be used for both acute and chronic wounds. Categories for dressings in this section start with the least absorptive, moisture-donating hydrogel dressings, followed by increasingly more absorptive dressings. </p>",
			childTaxonomies: [
				{
					title: "Hydrogel dressings",
					slug: "hydrogel-dressings",
					text: "<p> Hydrogel dressings are most commonly supplied as an amorphous, cohesive topical application that can take up the shape of a wound.</p>",
					productGroups: [],
					childTaxonomies: [
						{
							title: "Sodium hyaluronate dressings",
							slug: "sodium-hyaluronate-dressings",
						},
						{
							title: "Hydrogel sheet dressings",
							slug: "hydrogel-sheet-dressings",
						},
						{
							title: "Hydrogel application (amorphous)",
							slug: "hydrogel-application-amorphous",
						},
					],
				},
				{
					title: "Vapour-permeable films and membranes",
					slug: "vapour-permeable-films-and-membranes",
					text: "<p> Vapour-permeable films and membranes allow the passage of water vapour and oxygen but are impermeable to water and micro-organisms, and are suitable for lightly exuding wounds.</p>",
					productGroups: [],
					childTaxonomies: [
						{
							title: "Non-woven fabric dressing with viscose-rayon pad",
							slug: "non-woven-fabric-dressing-with-viscose-rayon-pad",
						},
						{
							title:
								"Vapour–permeable transparent film dressing with adhesive foam border",
							slug: "vapour-permeable-transparent-film-dressing-with-adhesive-foam-border",
						},
						{
							title: "Vapour–permeable transparent, adhesive film dressing.",
							slug: "vapour-permeable-transparent-adhesive-film-dressing",
						},
						{
							title:
								"Vapour-permeable Adhesive Film Dressing (Semi-permeable Adhesive Dressing)",
							slug: "vapour-permeable-adhesive-film-dressing-semi-permeable-adhesive-dressing",
						},
					],
				},
				{
					title: "Capillary-acting dressings",
					slug: "capillary-acting-dressings",
					text: null,
					productGroups: [],
					childTaxonomies: [],
				},
			],
		},
	},
};

describe("Wound management taxonomy page", () => {
	it("should render the page title with the expected text", async () => {
		render(<WoundManagementTaxonomyPage {...props} />);
		await waitFor(() => {
			expect(document.title).toStartWith(
				props.data.bnfWoundManagementTaxonomy.title
			);
		});
	});

	it("should render meta description", async () => {
		render(<WoundManagementTaxonomyPage {...props} />);
		await waitFor(() => {
			// eslint-disable-next-line testing-library/no-node-access
			expect(document.querySelector("meta[name=description]")).toHaveAttribute(
				"content",
				`This wound management topic describes the options that are currently recommended for ${props.data.bnfWoundManagementTaxonomy.title}`
			);
		});
	});

	it("should match page snapshot", () => {
		render(<WoundManagementTaxonomyPage {...props} />);
		expect(screen.getByRole("main")).toMatchSnapshot();
	});

	it("should add content start skip link target id to page header", () => {
		render(<WoundManagementTaxonomyPage {...props} />);
		const heading1 = screen.getByRole("heading", {
			level: 1,
		});
		// eslint-disable-next-line testing-library/no-node-access
		expect(heading1.parentElement).toHaveProperty("id", "content-start");
	});

	it("should render the page heading with the expected text", () => {
		render(<WoundManagementTaxonomyPage {...props} />);
		expect(screen.getByRole("heading", { level: 1 })).toHaveTextContent(
			props.data.bnfWoundManagementTaxonomy.title
		);
	});

	it("should render a navigation list of sub sections", () => {
		render(<WoundManagementTaxonomyPage {...props} />);
		expect(
			screen.getByRole("navigation", {
				name: "Wound management product pages",
			})
		).toBeInTheDocument();
	});

	it("should render correct number of sections in the sub nav", () => {
		render(<WoundManagementTaxonomyPage {...props} />);
		const subNav = screen.getByRole("navigation", {
			name: "Wound management product pages",
		});
		expect(within(subNav).getAllByRole("listitem")).toHaveLength(
			props.data.allBnfWoundManagementTaxonomy.taxonomies.length
		);
	});

	it("should show a message when rendering a product group with no information", () => {
		const noInformationWarningText =
			"Please note, there is currently no specific information about this product.";

		render(<WoundManagementTaxonomyPage {...props} />);
		expect(screen.queryByText(noInformationWarningText)).toBeNull();

		render(
			<WoundManagementTaxonomyPage
				data={{
					...props.data,
					bnfWoundManagementTaxonomy: {
						...props.data.bnfWoundManagementTaxonomy,
						childTaxonomies: [
							{
								title: "No info test",
								slug: "no-info-test",
								text: "<p>No info test</p>",
								childTaxonomies: [],
								productGroups: [
									{
										title: "Test product group",
										description: "<p>Test description</p>",
										products: [], // Need a product group with no products to trigger this condition
									},
								],
							},
						],
					},
				}}
			/>
		);
		expect(screen.getByText(noInformationWarningText)).toBeInTheDocument();
	});
});
