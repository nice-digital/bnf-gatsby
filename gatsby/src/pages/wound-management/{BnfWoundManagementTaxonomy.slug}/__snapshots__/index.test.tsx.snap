// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wound management taxonomy page should match page snapshot 1`] = `
<main>
  <div
    class="container"
  >
    <nav
      aria-label="Breadcrumbs"
      role="navigation"
    >
      <ol
        class="breadcrumbs"
      >
        <li
          class="breadcrumbs__crumb"
        >
          <a
            href="https://www.nice.org.uk/"
          >
            NICE
          </a>
        </li>
        <li
          class="breadcrumbs__crumb"
        >
          <a
            href="/"
          >
            BNF
          </a>
        </li>
        <li
          class="breadcrumbs__crumb"
        >
          <a
            href="/wound-management/"
          >
            Wound management products and elasticated garments
          </a>
        </li>
        <li
          class="breadcrumbs__crumb"
        >
          <span>
            Advanced wound dressings
          </span>
        </li>
      </ol>
    </nav>
    <script
      type="application/ld+json"
    >
      {
  "@context": "http://schema.org",
  "@type": "BreadcrumbList",
  "itemListElement": [
    {
      "@type": "ListItem",
      "position": 1,
      "item": {
        "@id": "https://www.nice.org.uk/",
        "name": "NICE"
      }
    },
    {
      "@type": "ListItem",
      "position": 2,
      "item": {
        "@id": "/",
        "name": "BNF"
      }
    },
    {
      "@type": "ListItem",
      "position": 3,
      "item": {
        "@id": "/wound-management/",
        "name": "Wound management products and elasticated garments"
      }
    },
    {
      "@type": "ListItem",
      "position": 4,
      "item": {
        "name": "Advanced wound dressings"
      }
    }
  ]
}
    </script>
    <div
      class="page-header"
      id="content-start"
    >
      <h1
        class="page-header__heading"
      >
        <span>
          Advanced wound dressings
        </span>
      </h1>
    </div>
    <div
      class="grid grid--loose"
      data-testid="body"
    >
      <div
        class="hide-print"
        data-g="12 md:4 lg:3"
      >
        <nav
          aria-label="Wound management product pages"
          class="stacked-nav"
        >
          <p
            class="stacked-nav__root"
          >
            <a
              aria-current="false"
              href="/wound-management/"
            >
              Wound management products
            </a>
          </p>
          <ul
            class="stacked-nav__list"
          >
            <li
              class="stacked-nav__list-item"
            >
              <a
                aria-current="false"
                href="/wound-management/basic-wound-contact-dressings/"
              >
                <span
                  class="stacked-nav__content-wrapper"
                >
                  <span>
                    Basic wound contact dressings
                  </span>
                </span>
              </a>
            </li>
            <li
              class="stacked-nav__list-item"
            >
              <a
                aria-current="false"
                href="/wound-management/advanced-wound-dressings/"
              >
                <span
                  class="stacked-nav__content-wrapper"
                >
                  <span>
                    Advanced wound dressings
                  </span>
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
      <div
        data-g="12 md:8 lg:9"
      >
        <nav
          aria-labelledby="navigate-to-section"
          class="nav navReadableMaxWidth"
        >
          <h2
            class="heading"
            id="navigate-to-section"
          >
            Navigate to section
          </h2>
          <ol
            aria-label="Jump links to sections on this page"
            class="linkList"
          >
            <li>
              <a
                href="#hydrogel-dressings"
              >
                Hydrogel dressings
              </a>
            </li>
            <li>
              <a
                href="#vapour-permeable-films-and-membranes"
              >
                Vapour-permeable films and membranes
              </a>
            </li>
            <li>
              <a
                href="#capillary-acting-dressings"
              >
                Capillary-acting dressings
              </a>
            </li>
          </ol>
        </nav>
        <div>
          <p>
             Advanced wound dressings can be used for both acute and chronic wounds. Categories for dressings in this section start with the least absorptive, moisture-donating hydrogel dressings, followed by increasingly more absorptive dressings. 
          </p>
        </div>
        <ul
          aria-label="List of advanced wound dressings"
          class="childTaxonomyList"
        >
          <li>
            <h2
              id="hydrogel-dressings"
            >
              Hydrogel dressings
            </h2>
            <div>
              <p>
                 Hydrogel dressings are most commonly supplied as an amorphous, cohesive topical application that can take up the shape of a wound.
              </p>
            </div>
            <ul
              class="nestedTaxonomyList"
            >
              <li>
                <a
                  href="/wound-management/advanced-wound-dressings/sodium-hyaluronate-dressings/"
                >
                  Sodium hyaluronate dressings
                </a>
              </li>
              <li>
                <a
                  href="/wound-management/advanced-wound-dressings/hydrogel-sheet-dressings/"
                >
                  Hydrogel sheet dressings
                </a>
              </li>
              <li>
                <a
                  href="/wound-management/advanced-wound-dressings/hydrogel-application-amorphous/"
                >
                  Hydrogel application (amorphous)
                </a>
              </li>
            </ul>
          </li>
          <li>
            <h2
              id="vapour-permeable-films-and-membranes"
            >
              Vapour-permeable films and membranes
            </h2>
            <div>
              <p>
                 Vapour-permeable films and membranes allow the passage of water vapour and oxygen but are impermeable to water and micro-organisms, and are suitable for lightly exuding wounds.
              </p>
            </div>
            <ul
              class="nestedTaxonomyList"
            >
              <li>
                <a
                  href="/wound-management/advanced-wound-dressings/non-woven-fabric-dressing-with-viscose-rayon-pad/"
                >
                  Non-woven fabric dressing with viscose-rayon pad
                </a>
              </li>
              <li>
                <a
                  href="/wound-management/advanced-wound-dressings/vapour-permeable-transparent-film-dressing-with-adhesive-foam-border/"
                >
                  Vapour–permeable transparent film dressing with adhesive foam border
                </a>
              </li>
              <li>
                <a
                  href="/wound-management/advanced-wound-dressings/vapour-permeable-transparent-adhesive-film-dressing/"
                >
                  Vapour–permeable transparent, adhesive film dressing.
                </a>
              </li>
              <li>
                <a
                  href="/wound-management/advanced-wound-dressings/vapour-permeable-adhesive-film-dressing-semi-permeable-adhesive-dressing/"
                >
                  Vapour-permeable Adhesive Film Dressing (Semi-permeable Adhesive Dressing)
                </a>
              </li>
            </ul>
          </li>
          <li>
            <h2
              id="capillary-acting-dressings"
            >
              Capillary-acting dressings
            </h2>
            <ul
              class="nestedTaxonomyList"
            >
              <li>
                <a
                  href="/wound-management/advanced-wound-dressings/capillary-acting-dressings/"
                >
                  Capillary-acting dressings
                </a>
              </li>
            </ul>
          </li>
        </ul>
      </div>
    </div>
  </div>
</main>
`;
